src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -240,6 +240,27 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
                     {
                         Error(diagnostics, ErrorCode.ERR_AttributeCtorInParameter, node, attributeConstructor.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat));
                     }
+
+                    for (var i = 0; i < boundConstructorArguments.Length; i++)
+                    {
+                        var parameter = GetCorrespondingParameter(i, attributeConstructor.Parameters, argsToParamsOpt, expanded);
+                        if (parameter is null)
+                        {
+                            continue;
+                        }
+
+                        var parameterType = (expanded && parameter is { IsParams: true, Type: ArrayTypeSymbol { IsSZArray: true } arrayType }) ? arrayType.ElementType : parameter.Type;
+
+                        // An enum constant as an object attribute argument triggers serialization of the enum's type.
+                        // This would fail for enums nested in a type referencing a function pointer, because
+                        // function pointer serialization is not supported, see https://github.com/dotnet/roslyn/issues/48765.
+                        if (parameterType.IsObjectType() &&
+                            boundConstructorArguments[i] is BoundConversion conv &&
+                            conv.Operand.Type is { } type && type.IsEnumType() && type.ContainsFunctionPointer())
+                        {
+                            diagnostics.Add(ErrorCode.ERR_FunctionPointerTypesInAttributeNotSupported, boundConstructorArguments[i].Syntax.Location);
```

Comment:
>diagnostics.Add(ErrorCode.ERR_FunctionPointerTypesInAttributeNotSupported, boundConstructorArguments[i].Syntax.Location); [](http://example.com/codeflow?start=28&length=121)

Could you elaborate why this check was moved here?

Reply:
Previously this check was only inside `BindDefaultExpression` because I thought I only needed to handle `default(...)` expressions. But turns out I need to handle any constant value of enum type in attribute arguments. So here I check attribute constructor arguments. And there's one more check which handles attribute named arguments.